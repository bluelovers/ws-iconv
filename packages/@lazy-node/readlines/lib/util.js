"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.splitBufferByBuffer = exports.bufferIndexWith = exports.bufferIndexWithByBuffer = exports.bufferStripEndWith = exports.bufferStripEndWithByBuffer = exports.bufferEndWith = exports.bufferEndWithByBuffer = exports.handleOptionNewLineCharacter = void 0;
function handleOptionNewLineCharacter(newLineCharacter) {
    if (!newLineCharacter) {
        return [10 /* EnumNewLineCharacter.LF */];
    }
    else if (typeof newLineCharacter !== 'number') {
        let ls = [];
        for (let i = 0; i < newLineCharacter.length; i++) {
            let c = newLineCharacter[i];
            if (typeof c !== 'number') {
                ls.push(c.charCodeAt(0));
            }
            else {
                ls.push(c);
            }
        }
        return ls;
    }
    return [newLineCharacter];
}
exports.handleOptionNewLineCharacter = handleOptionNewLineCharacter;
function bufferEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWithByBuffer = bufferEndWithByBuffer;
function bufferEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[buf.length - 1] === value;
    }
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    return i !== -1 && (i + value.length) === buf.length;
}
exports.bufferEndWith = bufferEndWith;
function bufferStripEndWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
    if (i !== -1 && (i + value.length) === buf.length) {
        return buf.slice(0, i);
    }
    return buf;
}
exports.bufferStripEndWithByBuffer = bufferStripEndWithByBuffer;
function bufferStripEndWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        if (buf[buf.length - 1] === value) {
            return buf.slice(0, buf.length - 1);
        }
    }
    else {
        let i = buf.lastIndexOf(value, byteOffset !== null && byteOffset !== void 0 ? byteOffset : (0 - value.length), encoding);
        if (i !== -1 && (i + value.length) === buf.length) {
            /*
            console.dir({
                i,
                vl: value.length,
                bl: buf.length,

                v: value.toString(),
                b: buf.toString(),
                b2: buf.slice(0, i).toString(),
            })
             */
            return buf.slice(0, i);
        }
    }
    return buf;
}
exports.bufferStripEndWith = bufferStripEndWith;
function bufferIndexWithByBuffer(buf, value, byteOffset, encoding) {
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWithByBuffer = bufferIndexWithByBuffer;
function bufferIndexWith(buf, value, byteOffset, encoding) {
    if (typeof value === 'number') {
        return buf[byteOffset] === value;
    }
    let i = buf.indexOf(value, byteOffset, encoding);
    return i !== -1 && i === byteOffset;
}
exports.bufferIndexWith = bufferIndexWith;
function splitBufferByBuffer(buffer, value) {
    const lines = [];
    let bufferPosition = 0;
    let lastNewLineBufferPosition = 0;
    while (true) {
        let line;
        let bufferPositionValue = buffer[bufferPosition];
        if (bufferIndexWithByBuffer(buffer, value, bufferPosition++)) {
            line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
            lines.push(line);
            lastNewLineBufferPosition = bufferPosition;
        }
        else if (bufferPositionValue === undefined) {
            break;
        }
    }
    let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
    if (leftovers.length) {
        lines.push(leftovers);
    }
    return lines;
}
exports.splitBufferByBuffer = splitBufferByBuffer;
//# sourceMappingURL=util.js.map