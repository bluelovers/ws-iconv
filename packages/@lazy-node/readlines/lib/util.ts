import { INewLineCharacter, EnumNewLineCharacter } from './types';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		let ls = [] as number[]

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			let c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

export function bufferEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		return buf[buf.length - 1] === value
	}

	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	return i !== -1 && (i + value.length) === buf.length
}

export function bufferStripEndWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

	if (i !== -1 && (i + value.length) === buf.length)
	{
		return buf.slice(0, i)
	}

	return buf
}

export function bufferStripEndWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset?: number,
	encoding?: BufferEncoding,
)
{
	if (typeof value === 'number')
	{
		if (buf[buf.length - 1] === value)
		{
			return buf.slice(0, buf.length - 1);
		}
	}
	else
	{
		let i = buf.lastIndexOf(value, byteOffset ?? (0 - value.length), encoding)

		if (i !== -1 && (i + value.length) === buf.length)
		{
			/*
			console.dir({
				i,
				vl: value.length,
				bl: buf.length,

				v: value.toString(),
				b: buf.toString(),
				b2: buf.slice(0, i).toString(),
			})
			 */

			return buf.slice(0, i)
		}
	}

	return buf
}

export function bufferIndexWithByBuffer(buf: Buffer,
	value: Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function bufferIndexWith(buf: Buffer,
	value: string | number | Uint8Array,
	byteOffset: number,
	encoding?: BufferEncoding,)
{
	if (typeof value === 'number')
	{
		return buf[byteOffset] === value
	}

	let i = buf.indexOf(value, byteOffset, encoding)

	return i !== -1 && i === byteOffset
}

export function splitBufferByBuffer(buffer: Buffer, value: Uint8Array)
{
	const lines: Buffer[] = [];
	let bufferPosition = 0;

	let lastNewLineBufferPosition = 0;
	while (true)
	{
		let line: Buffer;
		let bufferPositionValue = buffer[bufferPosition];

		if (bufferIndexWithByBuffer(buffer, value, bufferPosition++))
		{
			line = buffer.slice(lastNewLineBufferPosition, bufferPosition += (value.length - 1));
			lines.push(line);
			lastNewLineBufferPosition = bufferPosition;
		}
		else if (bufferPositionValue === undefined)
		{
			break;
		}
	}

	let leftovers = buffer.slice(lastNewLineBufferPosition, bufferPosition);
	if (leftovers.length)
	{
		lines.push(leftovers);
	}

	return lines;
}
