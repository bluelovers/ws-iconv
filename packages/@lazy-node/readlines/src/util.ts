import { INewLineCharacter } from './types';
import { EnumNewLineCharacter } from './index';

export function handleOptionNewLineCharacter(newLineCharacter?: INewLineCharacter): number[]
{
	if (!newLineCharacter)
	{
		return [EnumNewLineCharacter.LF];
	}
	else if (typeof newLineCharacter !== 'number')
	{
		const ls = [] as number[];

		for (let i = 0; i < newLineCharacter.length; i++)
		{
			const c = newLineCharacter[i];

			if (typeof c !== 'number')
			{
				ls.push(c.charCodeAt(0))
			}
			else
			{
				ls.push(c)
			}
		}

		return ls
	}

	return [newLineCharacter]
}

