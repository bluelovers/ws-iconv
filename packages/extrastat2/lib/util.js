"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.reassemble = void 0;
const path_1 = require("path");
function reassemble(pathparts) {
    return (0, path_1.join)('/', ...pathparts);
}
exports.reassemble = reassemble;
//# sourceMappingURL=util.js.map