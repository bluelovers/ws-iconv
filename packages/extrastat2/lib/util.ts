import { join } from "path";

export function reassemble(pathparts: string[])
{
	return join('/', ...pathparts)
}
