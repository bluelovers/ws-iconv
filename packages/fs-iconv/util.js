"use strict";
/**
 * Created by user on 2019/3/17.
 */
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.trimFilename = void 0;
const sanitize_filename_1 = __importDefault(require("sanitize-filename"));
function trimFilename(name) {
    let ret = name.toString()
        .replace(/\r\n|\r|\n/g, ' ')
        .replace(/[\r\n\t  \xA0]+/g, ' ');
    return sanitize_filename_1.default(ret)
        .trim()
        .replace(/^[　\s_]+/g, '')
        .replace(/[　\s_]+$/g, '');
}
exports.trimFilename = trimFilename;
exports.default = exports;
//# sourceMappingURL=util.js.map