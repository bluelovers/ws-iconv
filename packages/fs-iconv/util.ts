/**
 * Created by user on 2019/3/17.
 */

import sanitize = require('sanitize-filename');

export function trimFilename(name): string
{
	let ret = name.toString()
		.replace(/\r\n|\r|\n/g, ' ')
		.replace(/[\r\n\t  \xA0]+/g, ' ')
	;

	return sanitize(ret)
		.trim()
		.replace(/^[　\s_]+/g, '')
		.replace(/[　\s_]+$/g, '')
		;
}

export default exports as typeof import('./util');
