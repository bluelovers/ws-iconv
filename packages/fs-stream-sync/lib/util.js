// @ts-ignore
Object.freeze(exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidXRpbC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbInV0aWwudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUEsYUFBYTtBQUNiLE1BQU0sQ0FBQyxNQUFNLENBQUMsT0FBTyxDQUFDLENBQUEiLCJzb3VyY2VzQ29udGVudCI6WyIvLyBAdHMtaWdub3JlXG5PYmplY3QuZnJlZXplKGV4cG9ydHMpXG4iXX0=