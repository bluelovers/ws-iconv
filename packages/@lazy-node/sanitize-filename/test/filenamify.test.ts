import { basename } from "path";
import sanitizeFilename from '../index';
import { inspect } from 'util';

/**
 * @see https://github.com/sindresorhus/filenamify/blob/main/test.js
 */
describe('filnamify()', () =>
{
	[
		'foo/bar',
		'foo//bar',
		'//foo//bar//',
		'foo\\\\\\bar',
		'////foo////bar////',
		'foo\u0000bar',
		'.',
		'..',
		'./',
		'../',
		'foo.bar.',
		'foo.bar..',
		'foo.bar...',
		'con',
		'foo/bar/nul',
		'c/n',
		'',
	].forEach(input =>
	{
		test(inspect(input), () =>
		{
			expect(sanitizeFilename(input)).toMatchSnapshot();

			expect(sanitizeFilename(input, {replacement: '🐴🐴'})).toMatchSnapshot();

			expect(sanitizeFilename(input, {replacement: 'con'})).toMatchSnapshot();

			expect(sanitizeFilename(input, {replacement: ''})).toMatchSnapshot();

			expect(sanitizeFilename(input, {replacement: ' '})).toMatchSnapshot();
		})
	})
});

test('length', () =>
{
	// Basename length: 152
	const filename = 'this/is/a/very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_long_filename.txt';

	expect(sanitizeFilename(basename(filename)))
		.toStrictEqual('very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_');

	expect(sanitizeFilename(basename(filename), { maxLength: 180 }))
		.toStrictEqual('very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_very_long_filename.txt');
});
