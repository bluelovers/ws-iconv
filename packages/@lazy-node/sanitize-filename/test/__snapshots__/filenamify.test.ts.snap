// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filnamify() '' 1`] = `""`;

exports[`filnamify() '' 2`] = `""`;

exports[`filnamify() '' 3`] = `""`;

exports[`filnamify() '' 4`] = `""`;

exports[`filnamify() '' 5`] = `""`;

exports[`filnamify() '.' 1`] = `"!"`;

exports[`filnamify() '.' 2`] = `"🐴🐴"`;

exports[`filnamify() '.' 3`] = `"con"`;

exports[`filnamify() '.' 4`] = `""`;

exports[`filnamify() '.' 5`] = `" "`;

exports[`filnamify() '..' 1`] = `"!"`;

exports[`filnamify() '..' 2`] = `"🐴🐴"`;

exports[`filnamify() '..' 3`] = `"con"`;

exports[`filnamify() '..' 4`] = `""`;

exports[`filnamify() '..' 5`] = `" "`;

exports[`filnamify() '../' 1`] = `"!"`;

exports[`filnamify() '../' 2`] = `"🐴🐴"`;

exports[`filnamify() '../' 3`] = `"con"`;

exports[`filnamify() '../' 4`] = `""`;

exports[`filnamify() '../' 5`] = `" "`;

exports[`filnamify() './' 1`] = `"!"`;

exports[`filnamify() './' 2`] = `"🐴🐴"`;

exports[`filnamify() './' 3`] = `"con"`;

exports[`filnamify() './' 4`] = `""`;

exports[`filnamify() './' 5`] = `" "`;

exports[`filnamify() '////foo////bar////' 1`] = `"foo!bar"`;

exports[`filnamify() '////foo////bar////' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() '////foo////bar////' 3`] = `"fooconbar"`;

exports[`filnamify() '////foo////bar////' 4`] = `"foobar"`;

exports[`filnamify() '////foo////bar////' 5`] = `"foo bar"`;

exports[`filnamify() '//foo//bar//' 1`] = `"foo!bar"`;

exports[`filnamify() '//foo//bar//' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() '//foo//bar//' 3`] = `"fooconbar"`;

exports[`filnamify() '//foo//bar//' 4`] = `"foobar"`;

exports[`filnamify() '//foo//bar//' 5`] = `"foo bar"`;

exports[`filnamify() 'c/n' 1`] = `"c!n"`;

exports[`filnamify() 'c/n' 2`] = `"c🐴🐴n"`;

exports[`filnamify() 'c/n' 3`] = `"cconn"`;

exports[`filnamify() 'c/n' 4`] = `"cn"`;

exports[`filnamify() 'c/n' 5`] = `"c n"`;

exports[`filnamify() 'con' 1`] = `"!"`;

exports[`filnamify() 'con' 2`] = `"🐴🐴"`;

exports[`filnamify() 'con' 3`] = `"con"`;

exports[`filnamify() 'con' 4`] = `""`;

exports[`filnamify() 'con' 5`] = `" "`;

exports[`filnamify() 'foo.bar.' 1`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar.' 2`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar.' 3`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar.' 4`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar.' 5`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar..' 1`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar..' 2`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar..' 3`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar..' 4`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar..' 5`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar...' 1`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar...' 2`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar...' 3`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar...' 4`] = `"foo.bar"`;

exports[`filnamify() 'foo.bar...' 5`] = `"foo.bar"`;

exports[`filnamify() 'foo//bar' 1`] = `"foo!bar"`;

exports[`filnamify() 'foo//bar' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() 'foo//bar' 3`] = `"fooconbar"`;

exports[`filnamify() 'foo//bar' 4`] = `"foobar"`;

exports[`filnamify() 'foo//bar' 5`] = `"foo bar"`;

exports[`filnamify() 'foo/bar' 1`] = `"foo!bar"`;

exports[`filnamify() 'foo/bar' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() 'foo/bar' 3`] = `"fooconbar"`;

exports[`filnamify() 'foo/bar' 4`] = `"foobar"`;

exports[`filnamify() 'foo/bar' 5`] = `"foo bar"`;

exports[`filnamify() 'foo/bar/nul' 1`] = `"foo!bar!nul"`;

exports[`filnamify() 'foo/bar/nul' 2`] = `"foo🐴🐴bar🐴🐴nul"`;

exports[`filnamify() 'foo/bar/nul' 3`] = `"fooconbarconnul"`;

exports[`filnamify() 'foo/bar/nul' 4`] = `"foobarnul"`;

exports[`filnamify() 'foo/bar/nul' 5`] = `"foo bar nul"`;

exports[`filnamify() 'foo\\\\\\\\\\\\bar' 1`] = `"foo!bar"`;

exports[`filnamify() 'foo\\\\\\\\\\\\bar' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() 'foo\\\\\\\\\\\\bar' 3`] = `"fooconbar"`;

exports[`filnamify() 'foo\\\\\\\\\\\\bar' 4`] = `"foobar"`;

exports[`filnamify() 'foo\\\\\\\\\\\\bar' 5`] = `"foo bar"`;

exports[`filnamify() 'foo\\x00bar' 1`] = `"foo!bar"`;

exports[`filnamify() 'foo\\x00bar' 2`] = `"foo🐴🐴bar"`;

exports[`filnamify() 'foo\\x00bar' 3`] = `"fooconbar"`;

exports[`filnamify() 'foo\\x00bar' 4`] = `"foobar"`;

exports[`filnamify() 'foo\\x00bar' 5`] = `"foo bar"`;
